File: src/client/java/com/somefrills/features/misc/FilterMessages.java
package com.somefrills.features.misc;

import com.somefrills.config.Feature;
import com.somefrills.config.FrillsConfig;
import com.somefrills.events.ChatMsgEvent;
import meteordevelopment.orbit.EventHandler;

import java.util.regex.Pattern;

public class FilterMessages extends Feature {

    private static final Pattern IMPLOSION_PATTERN =
            Pattern.compile("Your Implosion hit (\\d+) enem(?:y|ies) for ([\\d,.]+) damage\\.");

    public FilterMessages() {
        super(FrillsConfig.instance.misc.chatFilter.enabled);
    }

    private static boolean shouldFilter(String msg) {
        if (msg.contains("There are blocks in the way!")) {
            return true;
        }
        return IMPLOSION_PATTERN.matcher(msg).matches();
    }

    @EventHandler
    private void onChatMessage(ChatMsgEvent event) {
        if (shouldFilter(event.messagePlain)) {
            event.cancel();
        }
    }
}

File: src/main/kotlin/com/somefrills/features/misc/FilterMessages.kt
package com.somefrills.features.misc

import com.somefrills.config.FrillsMod

import com.somefrills.events.ChatMsgEvent
import com.somefrills.features.core.Feature
import com.somefrills.features.core.FrillsFeature
import meteordevelopment.orbit.EventHandler
import java.util.regex.Pattern

@FrillsFeature
object FilterMessages : Feature(FrillsMod.config.misc.chatFilter.enabled) {
    private val IMPLOSION_PATTERN: Pattern =
        Pattern.compile("Your Implosion hit (\\d+) enem(?:y|ies) for ([\\d,.]+) damage\\.")

    private fun shouldFilter(msg: String): Boolean {
        if (msg.contains("There are blocks in the way!")) {
            return true
        }
        return IMPLOSION_PATTERN.matcher(msg).matches()
    }

    @EventHandler
    private fun onChatMessage(event: ChatMsgEvent) {
        if (shouldFilter(event.plainMessage)) {
            event.cancel()
        }
    }
}